### ISSUE TITLE
When I ran the experiments in examples/translation, I encountered a problem

### ISSUE THREAD

**Issue by zjr35897** [pytorch/fairseq#1479]
I trained a new model on the IWSLT'14 German to English dataset just following: https://github.com/pytorch/fairseq/tree/master/examples/translation#iwslt14-german-to-english-transformer.
However, when i used generate.py to translate the test dataset using:
`fairseq-generate data-bin/iwslt14.tokenized.de-en \
    --path checkpoints/checkpoint_best.pt \
    --batch-size 128 --beam 5 --remove-bpe`
I just got the answer like this:
![image](https://user-images.githubusercontent.com/18071348/70504952-c8a11900-1b61-11ea-8f9b-3c3015844de0.png)
I just followed the tutorial，can you help me to solve this problem?
Thanks!

**Comment by huihuifan** [comment]
How long did you train the model for? Is it fully trained?

**Comment by zjr35897** [comment]
I had trained for 13 hours on V100

**Comment by zjr35897** [comment]
> How long did you train the model for? Is it fully trained?

![image](https://user-images.githubusercontent.com/18071348/70676762-30717400-1cc8-11ea-9073-730141df5b15.png)

**Comment by edunov** [comment]
@zjr35897  the PPL value you see in the end of training is way to high, something went wrong, can you paste your full training command here? 
Also it will be helpful if you redirect training log into a file and attach it here too.

**Comment by zjr35897** [comment]
> @zjr35897 the PPL value you see in the end of training is way too high, something went wrong, can you paste your full training command here?
> Also it will be helpful if you redirect training log into a file and attach it here too.

![image](https://user-images.githubusercontent.com/18071348/70682968-7edc3e00-1cdb-11ea-9630-d1bc8fbc1fdf.png)
This is my command.

**Comment by edunov** [comment]
Ok, can you please remove old checkpoints: 

rm -rf checkpoints/ 

and then run this command: 

 CUDA_VISIBLE_DEVICES=0 python train.py     data-bin/iwslt14.tokenized.de-en     --arch transformer_iwslt_de_en --share-decoder-input-output-embed     --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0     --lr 5e-4 --lr-scheduler inverse_sqrt --warmup-updates 4000     --dropout 0.3 --weight-decay 0.0001     --criterion label_smoothed_cross_entropy --label-smoothing 0.1     --max-tokens 4096 --no-progress-bar --log-interval 100 --max-epoch 50 | tee log.txt

then check if the model you get is good. If not, copy and paste the content of log.txt here: https://gist.github.com/   and send us the link. 

my guess is that you have some messed up checkpoints in your checkpoints/ folder (this is why you are at epoch 837) and your model doesn't train

**Comment by vnikouli** [comment]
I've encountered similar problem recently. I think that in default settings the model converges very slowly. You need to increase batch size and/or update_freq and/or learning rate.

Below are validation perplexity curves:
 
![devppl](https://user-images.githubusercontent.com/7334887/70695533-3bc1a100-1cc2-11ea-871e-7879f3111109.png)

 Blue line (default settings) 
```
CUDA_VISIBLE_DEVICES=0 fairseq-train \
    ${DATA_DIR}/data-bin/ \
    --arch transformer_iwslt_de_en --share-decoder-input-output-embed \
    --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0 \
    --lr 5e-4 --lr-scheduler inverse_sqrt --warmup-updates 4000 \
    --dropout 0.3 --weight-decay 0.0001 \
    --criterion label_smoothed_cross_entropy --label-smoothing 0.1 \
    --max-tokens 4096 --tensorboard-logdir ${TB_DIR} 
```
Red line ( default settings + update_freq 4)
```
CUDA_VISIBLE_DEVICES=0 fairseq-train \
    ${DATA_DIR}/data-bin/ \
    --arch transformer_iwslt_de_en --share-decoder-input-output-embed \
    --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0 \
    --lr 5e-4 --lr-scheduler inverse_sqrt --warmup-updates 4000 \
    --dropout 0.3 --weight-decay 0.0001 \
    --criterion label_smoothed_cross_entropy --label-smoothing 0.1 \
    --max-tokens 4096 --update-freq 4 --tensorboard-logdir ${TB_DIR}
```
Pink line 
```
CUDA_VISIBLE_DEVICES=0 python $FAIRSEQ/train.py ${DATA_DIR}/bin --save-dir 
${MODEL_DIR} -s en -t de --arch transformer_iwslt_de_en --lr-scheduler inverse_sqrt 
--optimizer adam --fp16 --dropout 0.3  --criterion label_smoothed_cross_entropy
 --label-smoothing 0.1 --clip-norm 0.0 --share-all-embeddings --keep-last-epochs 1
 --lr 0.001 --max-epoch 100 --max-tokens 9012 --update-freq 4 --tensorboard-logdir ${TB_DIR}
```

These are experiments for en-de, but colleague of mine had similar observation for other language pairs on iwslt dataset.

**Comment by edunov** [comment]
Oh, I didn't realize you're doing En->De, the original IWSLT setup was tuned for De->En so yes, if you switch direction you may need some tuning of dropout and learning rate (and update-freq may help too)

**Comment by zjr35897** [comment]
@edunov Thanks for your answer. I changed the new version of fairseq(0.9.0) to solve this problem. I used to train with fairseq version 0.7.0 for three days, with more than 800 epochs in total. I didn't know why it happened.

I trained in two directions containing En->De and De->en at the same time. And they both had the same problem before I used a version of fairseq.
